homeassistant/components/linkplay/media_player.py

```diff
@@ -252,6 +258,14 @@ async def async_play_media(
             media_id = play_item.url
 
         url = async_process_play_media_url(self.hass, media_id)
+
+        # Modify the the url if required to use IP address instead of hostname.
+        # This is required for compatibility with some devices.
+        if self._use_ip_url:
+            parsed_url = yarl.URL(url)
+            # Update the parsed URL with the local ip
+            url = parsed_url.with_host(self.hass.config.api.local_ip)
```

Comment:
Given your analysis in the issue thread, I assume this is only required when you want to play a local(-only) url that isn't externally exposed?

How does this work for music coming from a streaming provider? When the toggle is enabled, it will replace dns always with it's ip address?

Reply:
On first reading through the code I was under the impression that this function was only called with HA URL's. Upone re-reading I can see that this isn't a valid assumption, so I'll update the code to handle this case and update the PR.

Reply:
You can use this utility method: https://github.com/home-assistant/core/blob/9e04f618b81e4ff73053eb72e9439992dbff1c9f/homeassistant/helpers/network.py#L67

Reply:
Thanks for that, I'm still getting my head around how the options flow testing should work, will try and spend some time in the next couple of days and get that added.